docs/proposals/failover/application-failover.md

```diff
@@ -0,0 +1,566 @@
+If such a node is found, one or more lower priority Pods get evicted from the node. After the Pods are gone, the Pod with high priority can be scheduled.
+In that case, Pods with low priority may be evicted frequently and have fewer scheduling opportunities.
+The reason for preemption may be due to resource shortage, or it may be due to affinity between Pods.
+At this time, single-cluster failover cannot effectively solve the problem. Multi-cluster failover may be a reasonable way.
+
+In the multi-cluster scenario, user workloads may be deployed in multiple clusters to improve service high availability. 
+Karmada already supports multi-cluster failover when detecting a cluster fault. It's a consideration from a cluster perspective.
+However, the application may still be unavailable when the control plane of the cluster is in a healthy state. 
+Moreover, the reasons for the failure of each application are different. When a certain failure occurs in the cluster, not all applications need to be migrated.
+Therefore, Karmada needs to provide a means of fault migration from an application perspective.
+
+## Motivation
+
+The cluster administrators want to ensure that the application can run normally, regardless of which cluster the application is running. When the application is unavailable and it cannot self-heal within a single cluster,
+they want Karmada to reschedule the application and avoid clusters that failed to run before.
+
+## Goals
+
+- Extend the API of PropagationPolicy to provide users with the rescheduled configuration.
+- Extend the API of ResourceBinding to record previously evicted clusters.
+- Propose the implementation ideas for involved components, including `karmada-controller-manager`, `karmada-webhook` and `karmada-scheduler`.
+  For example, `karmada-controller-manager` need to add a component to observe the health status of the application. `Karmada-scheduler` need to
+  add a plugin to filter the evicted cluster.
+
+## Non-Goals
+
+## Proposal
+
+### User Stories (Optional)
+
+#### As a user, I deploy an application in multiple clusters with preemptive scheduling. When business peaks, low-priority applications cannot run normally for a long time.
+
+Supposing I deploy an application in a cluster with preemptive scheduling. When cluster resources are in short supply, low-priority applications that were running normally are preempted and cannot run normally for a long time.
+At this time, applications cannot self-heal within a cluster. Users wants to try to schedule it to another cluster to ensure that the service is continuously served and the replicas does not drop to zero.
+
+#### As a user, I deploy an application in multiple clusters for HA. I want to upgrade one of the clusters and migrate to the original cluster when a single point of failure occurs in the higher version cluster.
```

Comment:
I didn't get this story, i think if the cluster administrators want to upgrade the cluster, they might try to `drain` a cluster.

> and migrate to the original cluster

I didn't get what the `original cluster` means.

Reply:
`Drain` is a solution. However, the current mainstream cluster upgrade method does not need to migrate to existing services. For example, [CCE](https://support.huaweicloud.com/intl/zh-cn/usermanual-cce/cce_01_0197.html).